### DISCUSSION TITLE
Infer `input_names` in `mxnet.gluon.SymbolBlock.imports` API

### DISCUSSION THREAD

**Discussion by liusy182** [apache/incubator-mxnet#19568] [Q&A]
The API [mxnet.gluon.SymbolBlock.export](https://mxnet.apache.org/versions/1.7/api/python/docs/api/gluon/symbol_block.html#mxnet.gluon.SymbolBlock.export) exports input names by following the convention below according to documentation:

```
When there are only one input, it will have name data. When there Are more than one inputs, 
they will be named as data0, data1, etc.
```

The counterpart [mxnet.gluon.SymbolBlock.imports](https://mxnet.apache.org/versions/1.7/api/python/docs/api/gluon/symbol_block.html#mxnet.gluon.SymbolBlock.imports) requires the `input_names` to be specified as the second parameter.

This is problematic for clients who use this set of APIs. E.g. a client who have previously exported the model may not know the format of `input_names` when it tries to do import later. 

I see there are 2 solutions for this problem:
1. let the `mxnet.gluon.SymbolBlock.imports` being able to infer the `input_names`. (preferred)
2. have `mxnet.gluon.SymbolBlock.export` return the `input_names` which can later be used by the import.

Does gluon support any of the above functionality in anyways? If not, would it be a good idea to support this?

**Comment by szha** [comment]
@liusy182 thanks for bringing this up. I agree this is an issue to be solved. Between the two proposed approaches, I think having `imports` infer `input_names` seems like a good idea. To keep backward compatibility, I think we can set `imports`'s argument `input_names` to be optional and defaults to None.

**Reply by leezu** [reply]
Would you like to contribute a PR @liusy182?

**Reply by liusy182** [reply]
Sure. I would love to. What are the things to watch out on this area as someone new to the code base?

**Reply by szha** [reply]
You can check out the contribution guides here for tips: https://mxnet.apache.org/versions/master/community

**Reply by liusy182** [reply]
I have created #19569 to keep track.

I assume the inference should happen right before [here](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/block.py#L1658). What should be the C API to use to get the inferred parameters?
